extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/UnsourcedScrutinizer.java

```diff
@@ -23,23 +23,44 @@
  ******************************************************************************/
 package org.openrefine.wikidata.qa.scrutinizers;
 
-import org.wikidata.wdtk.datamodel.interfaces.EntityIdValue;
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.updates.ItemUpdate;
+import org.wikidata.wdtk.datamodel.interfaces.PropertyIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.Reference;
 import org.wikidata.wdtk.datamodel.interfaces.Statement;
 
+import java.util.List;
+
 /**
  * A scrutinizer checking for unsourced statements
  * 
  * @author Antonin Delpeuch
  *
  */
-public class UnsourcedScrutinizer extends StatementScrutinizer {
+public class UnsourcedScrutinizer extends EditScrutinizer {
 
-    public static final String type = "unsourced-statements";
+    public static final String CITATION_NEEDED_QID = "Q54554025";
+    public static final String generalType = "unsourced-statements";
+    public static final String existingItemType = "no-references-provided";
+    public static final String newItemType = "no-references-found";
 
     @Override
-    public void scrutinize(Statement statement, EntityIdValue entityId, boolean added) {
-        if (statement.getReferences().isEmpty() && added) {
-            warning(type);
+    public void scrutinize(ItemUpdate update) {
+        for (Statement statement : update.getAddedStatements()) {
+            PropertyIdValue pid = statement.getClaim().getMainSnak().getPropertyId();
+            List<Statement> constraintDefinitions = _fetcher.getConstraintsByType(pid, CITATION_NEEDED_QID);
+            List<Reference> referenceList = statement.getReferences();
+
+            if (referenceList.isEmpty()) {
+                if (!constraintDefinitions.isEmpty()) {
+                    QAWarning issue = new QAWarning(update.isNew() ? newItemType : existingItemType, pid.getId(), update.isNew() ? QAWarning.Severity.CRITICAL : QAWarning.Severity.INFO, 1);
```

Comment:
Why should the severity be"critical" for new items and "info" otherwise? "critical" is really high, it will prevent people from uploading anything with the tool. I do not think we should take into account whether the item is new or not here: there is no point uploading statements if we are sure that they already exist and are sourced in Wikidata.

Currently, we have the "warning" level for all unsourced statements. When there is an explicit "citation needed" constraint, we could perhaps just raise the severity by one level ("important")?